src/pudl_archiver/depositors/fsspec.py

```diff
@@ -0,0 +1,285 @@
+"""Depositor implementation using any fsspec compatible file-system as storage backend."""
+
+import logging
+import traceback
+from typing import BinaryIO
+
+import aiohttp
+from pydantic import BaseModel, ConfigDict
+from upath import UPath
+
+from ..frictionless import MEDIA_TYPES, DataPackage, Resource, ResourceInfo
+from ..utils import RunSettings, compute_md5
+from .depositor import (
+    DepositionAction,
+    DepositionChange,
+    DepositorAPIClient,
+    DraftDeposition,
+    PublishedDeposition,
+    register_depositor,
+)
+
+logger = logging.getLogger(f"catalystcoop.{__name__}")
+
+
+def _resource_from_upath(path: UPath, parts: dict[str, str]) -> Resource:
+    """Create a resource from a single file with partitions.
+
+    Args:
+        path: UPath pointing to resource on local or remote filesystem.
+        parts: Working partitions of current resource.
+    """
+    mt = MEDIA_TYPES[path.suffix[1:]]
+
+    return Resource(
+        name=path.name,
+        path=path.as_uri(),
+        remote_url=path.as_uri(),
+        title=path.name,
+        mediatype=mt,
+        parts=parts,
+        bytes=path.stat().st_size,
+        hash=compute_md5(path),
+        format=path.suffix,
+    )
+
+
+class Deposition(BaseModel):
```

Comment:
non-blocking - this is what represents the deposition *contents* as opposed to sort of the "logical state" of the deposition as a whole (`Draft/PublishedDeposition`), yes?